objects/DB_Poller.py

```diff
@@ -0,0 +1,74 @@
+import asyncio
+import pandas as pd
+import time
+import sys
+import os
+from .OPTSubprocess import OPTSubprocess
+from hyperopt.mongoexp import MongoTrials
+from hyperopt.exceptions import AllTrialsFailed
+
+sys.path.append('..')
+from config.HParams import MF5HParams
+
+class DB_Poller(OPTSubprocess):
+    def __init__(self, id, logger, cwd, ip, port, key, poll_interval):
+        super().__init__('DB_Poller', id, logger)
+        self.ip = ip
+        self.port = port
+        self.cwd = cwd
+        self.poll_interval = poll_interval
+        self.HParams = MF5HParams()
+        self.key = key
+        try:
+            os.mkdir(os.path.join(self.cwd, f'{self.key}_{self.HParams.filetype}s'))
+            self.log(f'Created {self.HParams.filetype} log directory', 0)
+        except FileExistsError:
+            pass
+
+    async def init_poller(self):
+        while True:
+            self.pullHParams()
+            await asyncio.sleep(self.poll_interval)
+
+    def pullHParams(self):
+        trials = MongoTrials(f'mongo://{self.ip}:{self.port}/db/jobs', exp_key=self.key)
+        # get best trial from run
+        try:
+            self.HParams.HParam2File(trials.best_trial.get('misc').get('vals').to_dict(), 
+                                     os.path.join(self.cwd, f'{self.key}_{self.HParams.filetype}s', f'best.{self.HParams.filetype}'))
+        except AllTrialsFailed:
+            pass
+        # get the rest of the trials
+        for i in range(len(trials.trials)):
+            self.HParams.HParam2File(trials.trials[i].get('misc').get('vals').to_dict(), 
+                                     os.path.join(self.cwd, f'{self.key}_{self.HParams.filetype}s', f'{self.HParams.filetype}_{i}.{self.HParams.filetype}'))
+
+        # get results of each trial and compile them into a list
+        results = []
+        minLoss = 9999999
```

Comment:
consider `np.inf` to be 100% sure you don't have a legit loss function greater than the initialization (I know, you won't......just style)

Reply:
100% Agree, make it look pretty :)